File: java/src/insert/insert.java
package insert;

import java.util.Arrays;

public class insert {
    public static void insertSort(int []nums){
        for (int i=1; i < nums.length; i++){
            int j = i-1;
            int num = nums[i];
            for (; j>=0 && nums[j]>num; j--){
                nums[j+1] = nums[j];
            }
            nums[j+1] = num;
        }
    }
    public static void main(String[] args) {
        int []nums = {2, 4, 6, 1, 5, 3};
        insertSort(nums);
        System.out.println(Arrays.toString(nums));
    }
}

File: c++/insert/insert.cpp
#include <iostream>
using namespace std;

void insertSort(int nums[], int size){
    for (int i=1; i<size; i++){
        int j = i-1;
        int num = nums[i];
        for (; j>=0 && nums[j]>num;j--){
            nums[j+1] = nums[j];
        }
        nums[j+1] = num;

    }
}

void printArray(int nums[], int size){
    for (int i=0; i<size; i++){
        cout<<nums[i]<<" ";
    }
    cout<<endl;
}

int main(){
    int nums[] = {3, 2, 5, 1, 6, 4};
    int size = 6;
    insertSort(nums, size);
    printArray(nums, size);
    system("pause");
    return 0;
}